src/Schedule.h

```diff
@@ -94,20 +94,66 @@ enum class PrefetchBoundStrategy {
  * function is done by generating a loop nest that spans its
  * dimensions. We schedule the inputs to that function by
  * recursively injecting realizations for them at particular sites
- * in this loop nest. A LoopLevel identifies such a site. */
+ * in this loop nest. A LoopLevel identifies such a site.
+ *
+ * A LoopLevel can be in one of two states:
+
+     - Unlocked (the default state): An unlocked LoopLevel can be mutated freely (via the set() method),
```

Comment:
I wonder if documenting the locking state is exposing a bit too much implementation detail. It seems to me that only things after lowering will be prevented from mutating the LoopLevel (i.e. no user code will ever be able to do this), and similarly, user code will never be able to inspect a LoopLevel, so we might simply make those APIs private.

The one exception to the above seems like it might be custom lowering passes, but I wonder if that should just get its own special notes, rather than complicating the documentation for the typical use case.

Reply:
after lowering -> during and after lowering

Reply:
> I wonder if documenting the locking state is exposing a bit too much implementation 

Yeah, I had the same thought. The flip is that if we don't document this, someone will try to use these methods (which are documented and public) and then be puzzled about why they all assert-fail. (Even if we make the assert-fail message much more explanatory, it's still gonna seem pretty weird.)